WalletWasabi.Gui/Tabs/WalletManager/LoadWallets/LoadWalletViewModel.cs

```diff
@@ -71,13 +71,9 @@ public LoadWalletViewModel(WalletManagerViewModel owner, LoadWalletType loadWall
 				.ObserveOn(RxApp.MainThreadScheduler)
 				.Subscribe(x => SelectedWallet = x);
 
-			LoadCommand = ReactiveCommand.Create(()=>
-			{
-				RxApp.MainThreadScheduler.Schedule(async () =>
-				{
-					await LoadWalletAsync();
-				});
-			}, this.WhenAnyValue(x => x.SelectedWallet, x => x?.WalletState).Select(x => x == WalletState.Uninitialized));
+			LoadCommand = ReactiveCommand.Create(() =>
+				RxApp.MainThreadScheduler.Schedule(async () => await LoadWalletAsync()),
```

Comment:
It turned out this is disposable. Can you add CompositDisposables property and add this and RootList to it and dispose the CompositDisposables in the Dispose method?

Reply:
@yahiheb this is what I meant. Every time you initialize a new object which is IDisposable you should consider adding the disposal of it. If the disposal is missing you will get a memory leak, if the disposal is there nothing happens. 
Rx has a different approach to this: you only have to dispose of if you are subscribing to a non-class member object. In this case, we might not need to dispose of the IDisposable created by Schedule but I am not sure.

https://stackoverflow.com/questions/58026600/should-i-dispose-ischeduler-schedule-subscription-mannualy-to-avoid-memory-leaks

@danwalmsley I got some ideas after reading that. How about adding RxApp.TaskpoolScheduler to the ReactiveCommand and remove the Schedule?

Reply:
@molnard Thank you very much.
I appreciate your explanation and your help.